### PR TITLE
Use the refactored read_crypto_test_vectors()

### PR DESCRIPTION
**Pull request by lmctv**
Janitorial stuff, but I love removing duplicated functions.

### CODE REVIEW

File: tests/test_signing.py [hunk 1]
```diff
@@ -34,23 +32,19 @@ def tohex(b):
...
-                    binascii.unhexlify(x[3].encode("ascii"))[:64],
-                ),
-            })
-
-    return answers
+    # hex-encoded fields on each input line: sk||pk, pk, msg, signature||msg
+    # known answer fields: sk, pk, msg, signature, signed
+    DATA = "ed25519"
+    lines = read_crypto_test_vectors(DATA, delimiter=b':')
+    answ = [(
```
Comment (reaperhulk) [inline tests/test_signing.py]: let's just call return here

Comment (lmctv) [reply tests/test_signing.py]: Corrected.

File: tests/utils.py [hunk 2]
```diff
@@ -27,14 +27,14 @@ def assert_not_equal(x, y):
     assert not (x == y)
 
 
-def read_crypto_test_vectors(fname, maxels=0):
+def read_crypto_test_vectors(fname, maxels=0, delimiter=b'\t'):
```
Comment (reaperhulk) [inline tests/utils.py]: Sorry one more comment: could we just update the call sites so we don't need a default value for delimiter here?

Comment (lmctv) [reply tests/utils.py]: I agree with updating callers, but I'd rather keep delimiter as a keyword argument. Update coming.

Comment (lmctv) [reply tests/utils.py]: Since #317 adds another caller, the one getting last must be rebased on the other.

Comment (reaperhulk) [reply tests/utils.py]: Let's land this refactor PR with the update and the other one can be rebased.

Comment (lmctv) [reply tests/utils.py]: Just pushed the update.

**Pull request merged**